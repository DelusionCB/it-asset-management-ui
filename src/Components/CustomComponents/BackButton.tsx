import React from 'react';
import {Button} from 'reactstrap';
import {useNavigate} from 'react-router-dom';
import {useTranslation} from 'react-i18next';
import './index.scss'

function Back (): JSX.Element {
    const navigate = useNavigate();
    const {t} = useTranslation()

    return (
        <Button className='back-button' onClick={() => { navigate(-1); }}>
            <i className='bi bi-arrow-left' />
            {t('back')}
        </Button>
    );
}

export default Back;
