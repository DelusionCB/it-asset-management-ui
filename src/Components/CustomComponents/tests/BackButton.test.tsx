import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import BackButton from '../BackButton';
import {BrowserRouter} from 'react-router-dom';
import '@testing-library/jest-dom'

describe('<BackButton />', () => {
    let renderer: TestRenderer.ReactTestRenderer;

    // Render the component before tests
    beforeAll(() => {
        renderer = TestRenderer.create(
            <BrowserRouter>
                <BackButton />,
            </BrowserRouter>,
        );
    });
    describe('Snapshot', () => {
        it('renders correctly', () => {
            expect(renderer.toJSON()).toMatchSnapshot();
        })
    })
})
