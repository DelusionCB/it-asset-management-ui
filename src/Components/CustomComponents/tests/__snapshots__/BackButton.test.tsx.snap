// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BackButton /> Snapshot renders correctly 1`] = `
[
  <button
    className="back-button btn btn-secondary"
    onClick={[Function]}
    type="button"
  >
    <i
      className="bi bi-arrow-left"
    />
    back
  </button>,
  ",",
]
`;
